pkg/ansible/proxy/proxy.go

```diff
@@ -326,16 +334,37 @@ func addWatchToController(owner metav1.OwnerReference, cMap *ControllerMap, reso
 		Version: gv.Version,
 		Kind:    owner.Kind,
 	}
-	c, watch, ok := cMap.Get(gvk)
+	c, watchDependentResources, watchClusterScopedResources, ok := cMap.Get(gvk)
 	if !ok {
 		return errors.New("failed to find controller in map")
 	}
 	u := &unstructured.Unstructured{}
 	u.SetGroupVersionKind(gvk)
 	// Add a watch to controller
-	if watch {
-		log.Info("Watching child resource", "kind", resource.GroupVersionKind(), "enqueue_kind", u.GroupVersionKind())
-		err = c.Watch(&source.Kind{Type: resource}, &handler.EnqueueRequestForOwner{OwnerType: u})
+	if watchDependentResources {
+		// Use EnqueueRequestForOwner unless user has configured watching cluster scoped resources
+		if useOwnerReference && !watchClusterScopedResources {
+			log.Info("Watching child resource", "kind", resource.GroupVersionKind(), "enqueue_kind", u.GroupVersionKind())
+			err = c.Watch(&source.Kind{Type: resource}, &handler.EnqueueRequestForOwner{OwnerType: u})
+		} else if watchClusterScopedResources {
+			// Use Map func since EnqueuRequestForOwner won't work
+			log.Info("Watching child resource which is cluster-scoped", "kind", resource.GroupVersionKind(), "enqueue_kind", u.GroupVersionKind())
+			watchMap[owner.UID] = owner
+			err = c.Watch(
+				&source.Kind{Type: resource},
+				&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
+					obj := watchMap[owner.UID]
```

Comment:
I think this would allow for resources that are not owned by the resource to trigger reconcile requests. We should be using the `MapObject` to find the correct owner we care about and then send the reconcile. If we don't find the owner ref we care about then we should ignore this and not send any requests.

Reply:
Yeah I completely lost sight of what I was doing with `watchMap` I am fixing this I know what I was doing wrong. I need to store a map of owner UID to created resources. Then we can deep copy the object in the map function to determine if it is in the list of created resources.